packages/expo-router/jest.config.js

```diff
@@ -15,12 +16,36 @@ function withDefaults({ watchPlugins, ...config }) {
   };
 }
 
-module.exports = withWatchPlugins({
-  projects: [
-    // Create a new project for each platform.
-    getWebPreset(),
-    getNodePreset(),
-    getIOSPreset(),
-    getAndroidPreset(),
-  ].map(withDefaults),
+const projects = [
+  // Create a new project for each platform.
+  getWebPreset(),
+  getNodePreset(),
+  getIOSPreset(),
+  getAndroidPreset(),
+].map(withDefaults);
+
+projects.push(
+  {
+    displayName: { name: 'Types', color: 'blue' },
+    runner: 'jest-runner-tsd',
+    testMatch: ['<rootDir>/src/typed-routes/__tests__/*.tsd.ts'],
+    rootDir: path.resolve(__dirname),
+    roots: ['src'],
+    globalSetup: '<rootDir>/src/typed-routes/testSetup.ts',
+  },
+  {
+    displayName: { name: 'Type Generation', color: 'blue' },
+    testMatch: ['<rootDir>/src/typed-routes/__tests__/*.node.ts'],
+    rootDir: path.resolve(__dirname),
+    roots: ['src'],
+    clearMocks: true,
+  }
+);
+
+const config = withWatchPlugins({
+  projects,
 });
+
+config.watchPlugins = [];
```

Comment:
Need to resolve this before merging. Jest is failing when I try to add additional projects?